File: 11기 코딩테스트_230308/연습2/MostAppearLetter/MostAppearLetter_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.MostAppearLetter.MostAppearLetter import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        s = 'google'

        # when
        result = solution(s)

        # then
        self.assertEqual('g', result)

    def test_2(self):
        # given
        s = 'algorithm'

        # when
        result = solution(s)

        # then
        self.assertEqual('a', result)

    def test_3(self):
        # given
        s = 'ajseifnaslgksadkjfsdjfklskfjlsajlf'

        # when
        result = solution(s)

        # then
        self.assertEqual('s', result)

    def test_4(self):
        # given
        s = 'zfa'

        # when
        result = solution(s)

        # then
        self.assertEqual('a', result)

    def test_5(self):
        # given
        s = 'a'

        # when
        result = solution(s)

        # then
        self.assertEqual('a', result)

    def test_6(self):
        # given
        s = 'z'

        # when
        result = solution(s)

        # then
        self.assertEqual('z', result)


if __name__ == '__main__':
    unittest.main()

File: 11기 코딩테스트_230308/연습2/MostAppearLetter/MostAppearLetter_test.js
import assert from 'assert'
import solution from './MostAppearLetter.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const s = 'google'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('g', result)
    done();
  });

  it('test 2', function (done) {
    // given
    const s = 'algorithm'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('a', result)
    done();
  });

  it('test 3', function (done) {
    // given
    const s = 'ajseifnaslgksadkjfsdjfklskfjlsajlf'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('s', result)
    done();
  });

  it('test 4', function (done) {
    // given
    const s = 'azbbddddbbza'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('b', result)
    done();
  });

  it('test 5', function (done) {
    // given
    const s = 'a'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('a', result)
    done();
  });

  it('test 6', function (done) {
    // given
    const s = 'z'

    // when
    const result = solution(s)

    // then
    assert.deepStrictEqual('z', result)
    done();
  });
})
